apps/myaccount/src/api/validation.ts

```diff
@@ -147,15 +147,28 @@ export const getUsernameConfiguration = (configs: ValidationDataInterface[]): Va
     const usernameConf: ValidationDataInterface[] =
         configs?.filter((data: ValidationDataInterface) => data.field === "username");
 
-    if (usernameConf === undefined || usernameConf.length < 1) {
-        return;
+    if (usernameConf === undefined || usernameConf.length < 1
+        || usernameConf[0].rules === undefined || usernameConf[0].rules.length < 1) {
+        return {
+            enableValidator: "false",
+            field: "username",
+            maxLength: 255,
+            minLength: 3,
+            type: "rules"
+        };
     }
 
     const config: ValidationDataInterface = usernameConf[0];
     const rules: ValidationConfInterface[] = config?.rules;
 
     if (rules?.length < 1) {
```

Comment:
`usernameConf[0].rules.length < 1)` => we checked the same with this 151. Do we need it here?

Reply:
Addressed by 33db5f901892e1c432cc5f517d94d2d925cd49ec.